source/postprocess/particle_distribution_statistics.cc

```diff
@@ -0,0 +1,397 @@
+/*
+  Copyright (C) 2025 - by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+#include <aspect/postprocess/particle_distribution_statistics.h>
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    template <int dim>
+    std::pair<std::string,std::string>
+    ParticleDistributionStatistics<dim>::execute (TableHandler &statistics)
+    {
+      unsigned int cells_with_particles = 0;
+      double standard_deviation_sum = 0;
+      double standard_deviation_min = std::numeric_limits<double>::max();
+      double standard_deviation_max = std::numeric_limits<double>::min();
+      double function_min_sum = 0;
+      double function_max_sum = 0;
+      double function_min_min = std::numeric_limits<double>::max();
+      double function_max_max = std::numeric_limits<double>::min();
+
+
+      for (const auto &cell : this->get_dof_handler().active_cell_iterators())
+        {
+          if (cell->is_locally_owned())
+            {
+              unsigned int particles_in_cell = 0;
+              for (unsigned int particle_manager_index = 0; particle_manager_index < this->n_particle_managers(); ++particle_manager_index)
+                particles_in_cell += this->get_particle_manager(particle_manager_index).get_particle_handler().n_particles_in_cell(cell);
+
+              if (particles_in_cell > 0)
+                {
+                  ++cells_with_particles;
+                  ParticlePDF<dim> pdf(granularity);
+                  pdf.set_is_defined_per_particle(KDE_per_particle);
```

Comment:
this could be simpler as:
```suggestion
                  if (KDE_per_particle == false)
                    ParticlePDF<dim> pdf(granularity);
                  else
                    ParticlePDF<dim> pdf;
```
then you do not need the function `pdf.set_is_defined_per_particle`